packages/compiler-core/src/transforms/transformElement.ts

```diff
@@ -672,3 +673,7 @@ function stringifyDynamicPropNames(props: string[]): string {
   }
   return propsNamesString + `]`
 }
+
+function isComponentTag(tag: string) {
+  return tag[0].toLowerCase() + tag.slice(1) === 'component'
```

Comment:
isn't it faster to do a check like

```suggestion
  return tag === 'component' || tag === 'Component'
```

Since it's the compiler, I think it makes more sense. Size-wise it should be pretty much the same